### PR TITLE
Add transpose frontends.torch

### PR DESCRIPTION
**Pull request by PatternFinder**
issue: #2703

### CODE REVIEW

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 1]
```diff
@@ -26,3 +26,14 @@ def swapdims(input, dim0, dim1):
     "uint32",
     "uint64",
 )
+
+
+def transpose(input, dim0, dim1):
+    return ivy.swapaxes(input, dim0, dim1)
+
+
+transpose.unsupported_dtypes = (
```
Comment (jkeane508) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: Are you sure that uint dtypes cannot be used for this function?

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 2]
```diff
@@ -26,3 +26,14 @@ def swapdims(input, dim0, dim1):
     "uint32",
     "uint64",
 )
+
+
+def transpose(input, dim0, dim1):
+    return ivy.swapaxes(input, dim0, dim1)
```
Comment (jkeane508) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:32]: We would use `ivy.matrix_transpose` here not swap axes

**Review by jkeane508 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 1 above)
Comment (PatternFinder) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: according to ivy.swapaxes, the uint16, uint32, uint64 does seem to be unsupported

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 2 above)
Comment (jkeane508) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:32]: Having looked at the docs, I would agree.

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 1 above)
Comment (jkeane508) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: Are you sure? I've just ran it locally and it is accepting arrays with uint dtypes.

Comment (PatternFinder) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: This is the definition from `(https://github.com/unifyai/ivy/tree/master/ivy/functional/backends/torch)/manipulation.py`
it says `swapaxes.unsupported_dtypes` doesn't support uint types; did I misunderstood this or

```
def swapaxes(
    x: torch.Tensor, axis0: int, axis1: int, *, out: Optional[torch.Tensor] = None
) -> torch.Tensor:
    ret = torch.transpose(x, axis0, axis1)
    return ret


swapaxes.unsupported_dtypes = (
    "uint16",
    "uint32",
    "uint64",
)
```

Comment (jkeane508) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: Yep that is the backend PyTorch implementation of `swapaxes`. However, the frontend implementation is independent of this and `ivy.swapaxes()` has been designed as to accept any relevant dtype regardless of the framework. However, I have raised this with the team and will get back with an answer!

**Review by jkeane508 — CHANGES_REQUESTED** [review]
As mentioned above :)

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 1 above)
Comment (PatternFinder) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:35]: ok got it. so what's our policy here regarding the `frontend_function.unsupported_dtypes = ()`; should I keep including them in the frontend or not include them?

**Comment by PatternFinder** [comment]
> As mentioned above :)

ok I removed the `unsupported_types`; what else should I do for this PR? :)

**Review by jkeane508 — APPROVED** [review]
You can just resolve conflicts and provided everything is passing, we can merge :)

**Comment by PatternFinder** [comment]
ok resolved conflicts

**Pull request merged**